File: src/1071-Greatest-Common-Divisor-of-Strings/1071.js
var gcdOfStrings = function(s1, s2) {
    let gcd = function(a, b) {
        return b ? gcd(b, a % b) : a;
    }
    return s1 + s2 == s2 + s1 ? s1.substring(0, gcd(s1.length, s2.length)) : "";
};

File: src/1071-Greatest-Common-Divisor-of-Strings/1071.java
class Solution 
{
    public String gcdOfStrings(String s1, String s2) 
    {
        return (s1 + s2).compareTo(s2 + s1) == 0 ? s1.substring(0, gcd(s1.length(), s2.length())) : "";
    }
    
    private int gcd(int a, int b)  {
        return b > 0 ? gcd(b, a % b) : a;
    }
}